src/app/LoginPanel.tsx

```diff
@@ -25,22 +30,46 @@ export default function LoginPanel() {
 
   async function finishLogin(token: string) {
     const res = await Api.post<any>('/api/auth/google', { token }) as AuthResponse;
-    localStorage.userAuth = JSON.stringify(res.userInfo);
-
     console.log('login response', res);
-    dispatch(AuthActions.set({ userInfo: res.userInfo }));
-    dispatch(OrgActions.set({ mine: res.organization }))
+
+    if (res.userInfo) {
+      localStorage.userAuth = JSON.stringify(res.userInfo);
+      dispatch(AuthActions.set({ userInfo: res.userInfo }));
+      dispatch(OrgActions.set({ mine: res.organization }));
+    } else {
+      switch (res.error) {
+        case AuthError.USER_NOT_KNOWN:
+          setError("Could not find email address in D4H");
+          setErrorDetails("Please contact your unit's database manager to have your email address added to your D4H profile, or log in with an email address in your profile.");

```

Comment:
Nit - it's not a given that the organization is using D4H as their membership store.

Reply:
Hm. Good point. I don't know if there's a way to word this without using "D4H" (or whatever the provider is) while keeping it directly useful to users.

Can pass the member provider name through site config, but then that violates the principle of keeping org data off the client when the user is unauthenticated.

Reply:
Going to get this in as-is for the demo. Given our discussions in Slack, I'm not concerned about leaking that D4H is in use since the error message will only appear after oauth succeeds.

I'll fast follow with a change to send down the org info for these error messages and customize the error text based on that info.